futures-mutex/src/lib.rs

```diff
@@ -82,16 +31,15 @@ unsafe impl<T: Send> Sync for Inner<T> {}
 /// *As of now, there is no strong guarantee that a particular handle of the lock won't be starved. Hopefully the use of the queue will prevent this, but I haven't tried to test that.*
 #[derive(Debug)]
 pub struct Mutex<T> {
-    inner: Arc<Inner<T>>
+    inner: Arc<Inner<T>>,
```

Comment:
I know that the original library had this, but can we drop the `Arc<Inner>` and make plain `Inner` the `Mutex`? The stdlib `Mutex` requires `Arc<Mutex>` for shared ownership so this would keep it consistent. Plus, `Inner` is `Sync` so removing the `Arc` would allow constructing `Mutex` in a `lazy_static!` and sharing it across threads zero-cost.